src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultParameterValidator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class DefaultParameterValidator : IParameterValidator
```

Comment:
What if we wrapped up both the model binding and validation responsibilities into a single interface `IBinder`. I can't think of a case where you'd want to separate them...

The idea is that I binder would close over details like the MM, and parameter/property descriptor. 

Basically take the code you have here and consider pushing it up a level.

Reply:
As suggested, have pushed this up a level. Have not introduced a new `IBinder` extensibility point though - we already use the concrete class `ParameterBinder` as in your other comment below.